sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -530,211 +376,99 @@ class wi_element<uint16_t, NumRows, NumCols, Layout, Group> {
     return (uint16_t)*res;
   }
 
-  friend uint16_t
-  operator+(const wi_element<uint16_t, NumRows, NumCols, Layout, Group> &lhs,
-            const uint16_t &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return make_bf16(
-        make_fp32(__spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx)) +
-        make_fp32(rhs));
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  wi_element &operator+=(const uint16_t &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    M.spvm = __spirv_VectorInsertDynamic(
-        M.spvm,
-        make_bf16(make_fp32(__spirv_VectorExtractDynamic(M.spvm, idx)) +
-                  make_fp32(rhs)),
-        idx);
-    return *this;
-#else
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  friend uint16_t
-  operator-(const wi_element<uint16_t, NumRows, NumCols, Layout, Group> &lhs,
-            const uint16_t &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return make_bf16(
-        make_fp32(__spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx)) -
-        make_fp32(rhs));
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
+#if __SYCL_DEVICE_ONLY__
+#define OP(opassign, op)                                                       \
+  wi_element &operator opassign(const uint16_t &rhs) {                         \
+    M.spvm = __spirv_VectorInsertDynamic(                                      \
+        M.spvm,                                                                \
+        make_bf16(make_fp32(__spirv_VectorExtractDynamic(M.spvm, idx)          \
+                                op make_fp32(rhs))),                           \
+        idx);                                                                  \
+    return *this;                                                              \
+  }
+#else // __SYCL_DEVICE_ONLY__
+#define OP(opassign, op)                                                       \
+  wi_element &operator opassign(const uint16_t &rhs) {                         \
+    (void)rhs;                                                                 \
+    throw runtime_error("joint matrix is not supported on host device.",       \
+                        PI_INVALID_DEVICE);                                    \
+  }
+#endif // __SYCL_DEVICE_ONLY__
+  OP(+=, +)
+  OP(-=, -)
+  OP(*=, *)
+  OP(/=, /)
+#undef OP
+
+#if __SYCL_DEVICE_ONLY__
+#define OP(type, op)                                                           \
```

Comment:
It would be preferable to have a more informative name for the macro to distinguish it from the other, even if they don't overlap. You could, for example name this this `BINOP` and the following relational operator macro `RELOP` (or `BINRELOP` if you prefer.) Just suggestions for the names, so feel free to change as you see fit.